strings/base_string.h

```diff
@@ -423,13 +423,13 @@ namespace winrt::impl
         hstring_builder& operator=(hstring_builder const&) = delete;
 
         explicit hstring_builder(uint32_t const size) :
-            m_handle(impl::precreate_hstring_on_heap(size))
+            m_handle(size ? impl::precreate_hstring_on_heap(size) : nullptr)
```

Comment:
It doesn't make sense to create an empty `hstring_builder` - it's internal so I'm ok with this for now but it would seem to invite misuse. This class is also only used by the `concat_hstring` internal helper so it would seem like a better fix would not be too difficult.